From 674975040bdd852679efb20e0762335927b2e223 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Tue, 24 Apr 2018 11:15:19 +0800
Subject: [PATCH 3688/5242] MLK-18123-2 gpu: imx: imx8_dprc: Make enable
 argument boolean for sel configure

commit  5a040a7aaba00c0f84170283eabf2808855ba9ed from
https://source.codeaurora.org/external/imx/linux-imx.git

The dprc_prg_sel_configure() function gets an enable argument.
Improve the enable argument type from 'unsigned int' to 'bool'.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/imx8_dprc.c |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/imx/imx8_dprc.c b/drivers/gpu/imx/imx8_dprc.c
index bd32c31..536d41b 100644
--- a/drivers/gpu/imx/imx8_dprc.c
+++ b/drivers/gpu/imx/imx8_dprc.c
@@ -330,8 +330,7 @@ static void dprc_dpu_gpr_configure(struct dprc *dprc, unsigned int stream_id)
 	sc_ipc_close(mu_id);
 }
 
-static void dprc_prg_sel_configure(struct dprc *dprc,
-				   u32 resource, unsigned int enable)
+static void dprc_prg_sel_configure(struct dprc *dprc, u32 resource, bool enable)
 {
 	sc_err_t sciErr;
 	sc_ipc_t ipcHndl = 0;
@@ -419,7 +418,7 @@ void dprc_configure(struct dprc *dprc, unsigned int stream_id,
 		} else {
 			if (dprc->sc_resource == SC_R_DC_0_BLIT1) {
 				dprc_prg_sel_configure(dprc,
-						       SC_R_DC_0_BLIT0, 1);
+						       SC_R_DC_0_BLIT0, true);
 				prg_set_auxiliary(dprc->prgs[1]);
 			}
 		}
@@ -427,7 +426,7 @@ void dprc_configure(struct dprc *dprc, unsigned int stream_id,
 	} else {
 		if (dprc->sc_resource == SC_R_DC_0_BLIT0
 			&& dprc->devtype->has_fixup) {
-			dprc_prg_sel_configure(dprc, SC_R_DC_0_BLIT0, 0);
+			dprc_prg_sel_configure(dprc, SC_R_DC_0_BLIT0, false);
 			prg_set_primary(dprc->prgs[0]);
 		}
 
-- 
1.7.9.5

